### PR TITLE
Fix for DRILL-4759: Drill throwing array index out of bound exception…

### PR DESCRIPTION
**Pull request by ppadma**
… when reading a parquet file written by map reduce program

### CODE REVIEW

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java [hunk 1]
```diff
@@ -156,12 +156,18 @@ protected void readField(long recordsToReadInThisPass) {
       recordsReadInThisIteration = Math.min(pageReader.currentPageCount
           - pageReader.valuesRead, recordsToReadInThisPass - valuesReadInCurrentPass);
 
-      for (int i = 0; i < recordsReadInThisIteration; i++){
-        try {
-        valueVec.getMutator().setSafe(valuesReadInCurrentPass + i, pageReader.dictionaryValueReader.readLong());
-        } catch ( Exception ex) {
-          throw ex;
+      if (usingDictionary) {
```
Comment (sudheeshkatkam) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java]: Why is this check added only for this reader?

Comment (ppadma) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java]: This should ideally be fixed for all types and will be done in a later commit after sufficient testing. For now, just fixing for the type that this issue is reported for.

Comment (sudheeshkatkam) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java]: Okay. Can you add a unit test for this case?

Comment (parthchandra) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ParquetFixedWidthDictionaryReaders.java]: Fix is fine, but agree that a unit test should be added.

**Comment by ppadma** [comment]
updated the pull request with test case added.

**Comment by parthchandra** [comment]
Can you log a new JIRA to investigate and address issues with any other column types?
+1 for this change.

**Comment by ppadma** [comment]
updated the fix. Please review. In DictionaryBigIntReader, we need to call parent readField if dictionary encoding is not enabled for the page.

**Comment by parthchandra** [comment]
+1 for updated patch.

**Comment by ppadma** [comment]
The pull request was merged in commit: e371e18

**Pull request closed without merging**